File: Programmers/Level2/12945.cpp
#include <string>
#include <vector>

using namespace std;

int fibonacci(int k) {
    static int cache [100000] = { 0, 1, 1,};

    int first = (k >= 1 && cache[k - 1] != 0) ? cache[k - 1] : fibonacci(k - 1);
    int second = (k >= 2 && cache[k - 2] != 0) ? cache[k - 2] : fibonacci(k - 2);
    
    cache[k] = (first + second) % 1234567;
    return (first + second) % 1234567;
}

int solution(int n) {
    int answer = fibonacci(n);
    return answer;
}

File: Programmers/Level2/12945.js
function solution(n) {
  const dp = [0, 1, 1];

  for (let i = 2; i <= n; ++i) {
    dp[i] = (dp[i - 1] + dp[i - 2]) % 1234567;
  }

  return dp[n];
}
